File: C++/Easy/WrongAnswer/find-subsequence-of-length-k-with-the-largest-sum_8.cpp
class Solution {
public:
    vector<int> maxSubsequence(vector<int>& nums, int k) {
        
        int n = nums.size();
        vector<int> res(k);
        if (n == k) {
            return nums;
        }
        int top = -1;
        for (int i = 0; i < n; ++i) {
            while (top >= 0 && res[top] < nums[i] && n - i > k - top - 1) {
                --top;
            }
            if (top < k - 1) {
                res[++top] = nums[i];
            }
        }
        return res;
    }
};

File: Java/Easy/WrongAnswer/find-subsequence-of-length-k-with-the-largest-sum_8.java
class Solution {
    public int[] maxSubsequence(int[] nums, int k) {
        
        int n = nums.length;
        int[] ans = new int[k];
        if (n == 0) {
            return ans;
        }
        int i = 0;
        int j = 0;
        int sum = 0;
        int max = Integer.MIN_VALUE;
        int maxI = -1;
        int maxJ = -1;
        while (j < n) {
            sum += nums[j];
            if (j - i + 1 == k) {
                if (sum > max) {
                    max = sum;
                    maxI = i;
                    maxJ = j;
                }
                sum -= nums[i];
                i++;
            }
            j++;
        }
        for (int l = maxI; l <= maxJ; l++) {
            ans[l - maxI] = nums[l];
        }
        return ans;
    }
}